speechbrain/nnet/quaternion_networks/q_normalization.py

```diff
@@ -120,38 +121,36 @@ def forward(self, input):
                 dim=0,
             )
 
-            denominator = torch.sqrt(quat_variance + self.eps)
+            # Reciprocal sqrt was 8x faster in testing
+            denominator = torch.rsqrt(quat_variance + self.eps)
 
-            # x - mu / sqrt(var + e)
-            out = input / torch.cat(
+            # (x - mu) / sqrt(var + e)
+            out = delta * torch.cat(
                 [denominator, denominator, denominator, denominator],
                 dim=self.dim,
             )
 
             # Update the running stats
             if self.track_running_stats:
-                self.running_mean = (
-                    1 - exponential_average_factor
-                ) * self.running_mean + exponential_average_factor * mu.view(
-                    self.running_mean.size()
-                )
-
-                self.running_var = (
-                    1 - exponential_average_factor
-                ) * self.running_var + exponential_average_factor * quat_variance.view(
-                    self.running_var.size()
-                )
+                if self.num_batches_tracked == 1:
+                    self.running_mean = mu
+                    self.running_var = quat_variance
+                else:
+                    self.running_mean = (
+                        1 - exponential_average_factor
+                    ) * self.running_mean + exponential_average_factor * mu
+
+                    self.running_var = (
+                        (1 - exponential_average_factor) * self.running_var
+                        + exponential_average_factor * quat_variance
+                    )
         else:
-            q_var = torch.cat(
-                [
-                    self.running_var,
-                    self.running_var,
-                    self.running_var,
-                    self.running_var,
-                ],
+            denominator = torch.rsqrt(self.running_var + self.eps)
+            denominator = torch.cat(
```

Comment:
Wouldn't torch.repeat be faster?

Reply:
Yes it probably would be. I was just basing it off how the existing code was written, but I can go back and fix that this weekend.